client/server/pages/index.js

```diff
@@ -723,7 +723,7 @@ module.exports = function() {
 		// redirect logged-out tag pages to en.wordpress.com
 		app.get( '/tag/:tag_slug', function( req, res, next ) {
 			if ( ! req.context.isLoggedIn ) {
-				res.redirect( 'https://en.wordpress.com/tag/' + encodeURIComponent( req.params.tag_slug ) );
+				res.redirect( 'https://wordpress.com/tag/' + encodeURIComponent( req.params.tag_slug ) );
```

Comment:
If they're not logged in, they're already here and this will loop. I think you can drop the redirect entirely, as this should happen unless we start serving unlogged tag pages from Calypso.